### PR TITLE
Wire the client side rate limiter metric

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Expose the latency of the client-side rate-limiter.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/monitoring/client/prometheus/prometheus.go [hunk 1]
```diff
@@ -51,8 +60,9 @@ func init() {
 	prometheus.MustRegister(requestLatency)
 	prometheus.MustRegister(requestResult)
 	metrics.Register(metrics.RegisterOpts{
-		RequestLatency: &latencyAdapter{requestLatency},
-		RequestResult:  &resultAdapter{requestResult},
+		RequestLatency:     &latencyAdapter{requestLatency},
+		RateLimiterLatency: &latencyAdapter{rateLimiterLatency},
```
Comment (rthallisey) [inline pkg/monitoring/client/prometheus/prometheus.go:64]: I could be missing something, but I don't see where `RateLimiterLatency` gets populated with data.

Comment (rmohr) [reply pkg/monitoring/client/prometheus/prometheus.go:64]: That is done via the latencyAdapter methods.

Comment (rthallisey) [reply pkg/monitoring/client/prometheus/prometheus.go:64]: I know but who uses them?  I don't see `Observe` or `Increment` called in the code base.

Comment (rmohr) [reply pkg/monitoring/client/prometheus/prometheus.go:64]: Here it is: https://github.com/kubevirt/kubevirt/blob/main/vendor/k8s.io/client-go/rest/request.go#L598

Comment (rthallisey) [reply pkg/monitoring/client/prometheus/prometheus.go:64]: ah ok it comes from client-go, thanks

**Comment by rmohr** [comment]
@rthallisey could you have another look? Rebased it.

**Review by dhiller — APPROVED** [review]
/approve

**Comment by davidvossel** [comment]
@rmohr should we create an alert for this metric as well?

**Pull request merged**